fluid-music/src/vst2ToReaperObject.ts

```diff
@@ -0,0 +1,83 @@
+import { FluidPlugin, PluginType } from './plugin';
+import FluidIpcClient = require('./cybr/IpcClient');
+
+const rppp = require('rppp')
+const fluid = require('./cybr/index');
+
+/**
+ * Create a `ReaperVst` object from an existing plugin on the cybr instance.
+ *
+ * @param {FluidIpcClient} client A FluidIpcClient that is connected to a cybr instance
+ * @param {string} trackName - See documentation for fluid.audiotrack.select()
+ * @param {string} pluginName - See documentation for fluid.plugin.select()
+ * @param {string} pluginType
+ * @param {string} pluginId 
+ * 
+ * @returns {ReaperVst}
+ */
+export async function vst2ToReaperObject(client: FluidIpcClient, trackName: string, plugin: FluidPlugin, n: number, bpm: number) {
+
+  const cybrType = (plugin.pluginType === PluginType.unknown) ? undefined : plugin.pluginType;
+  const pluginName = plugin.pluginName;
+
+  const msg = [
+    fluid.audiotrack.select(trackName),
+    fluid.plugin.select(pluginName, cybrType, n),
+    fluid.plugin.getReport(),
+  ]
+
+  const retObj = await client.send(msg);
```

Comment:
I think `client.send` will fail silently if there is no active session. There are several possibilities for how to handle this:

1. Clearly document that `vst2ToReaperObject` can only be called with a `client` that has an active session, for example `cybr.global.activate('something.tracktionedit', true)` has already been called. With this option, the `tracksToReaperProject` should activate the session before processing any plugins.
2. Activate a new session on each call to `vst2ToReaperObject`.
3. solve the problem on the server side. For example, automatically activate an empty session if `/audiotrack/select` is received when no session is active.

Think about these, and decide which offers the best combination of user experience, and clean/simple implementation. Make sure to document it, because it should be clear that if the user has an active session when `vst2ToReaperObject` is called, it will be modified.

Reply:
I'm not sure if I understand this correctly, but the current code works for demo-typescript, which, as far as I know, does not activate its own session. I hope I am not misunderstanding what the code does.

Reply:
If it's working currently, can you double-check why it is working? It shouldn't be possible to select an audio track without an active session. See here:

https://github.com/CharlesHolbrow/cybr/blob/develop/Source/FluidOscServer.cpp#L610-L614

Reply:
If you start the server with `cybr -f` (without -e or -i) it will not have an active session. Try starting the server cleanly, and see if `demo-typescript` works. If it does, I think a session must be getting activated elsewhere. 

Once a session has been activated on the server, there is currently no way to "un-activate" it. So it could be that the reason that it's currently working there is a residual  active session from before running `demo-typescript/content.js`

Reply:
Ah, yes the reason was because I started cybr with -e. In this case, I think the best option would be the third option, because it would allow for tracksToReaperProject to be called without having to activate a session, and the second option would not be good because there is no way to "un-activate" a session, so we wouldn't want to risk memory problems on the server side. I'll push a change soon.